src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/collectionsgrid/viewmodel/ViewCollectionsViewModel.kt

```diff
@@ -0,0 +1,81 @@
+package org.wycliffeassociates.otter.jvm.app.ui.collectionsgrid.viewmodel
+
+import com.github.thomasnield.rxkotlinfx.observeOnFx
+import com.github.thomasnield.rxkotlinfx.toObservable
+import javafx.beans.property.SimpleBooleanProperty
+import javafx.beans.property.SimpleObjectProperty
+import javafx.beans.property.SimpleStringProperty
+import javafx.collections.FXCollections
+import javafx.collections.ObservableList
+import org.wycliffeassociates.otter.common.data.model.Collection
+import org.wycliffeassociates.otter.common.data.model.Content
+import org.wycliffeassociates.otter.common.domain.content.AccessTakes
+import org.wycliffeassociates.otter.jvm.app.ui.inject.Injector
+import tornadofx.*
+
+class ViewCollectionsViewModel: ViewModel() {
+    private val injector: Injector by inject()
+    private val directoryProvider = injector.directoryProvider
+    private val collectionRepository = injector.collectionRepo
+    private val contentRepository = injector.contentRepository
+    private val takeRepository = injector.takeRepository
+    private val pluginRepository = injector.pluginRepository
+
+    // The selected project
+    private var activeProject: Collection by property()
+    val activeProjectProperty = getProperty(ViewCollectionsViewModel::activeProject)
+
+    val projectTitle: SimpleStringProperty = SimpleStringProperty("")
+
+    // List of collection children (i.e. the chapters) to display in the list
+    var children: ObservableList<Collection> = FXCollections.observableList(mutableListOf())
+
+    // Selected child
+    private var activeCollection: Collection by property()
+    val activeCollectionProperty = getProperty(ViewCollectionsViewModel::activeCollection)
+
+    // List of content to display on the screen
+    // Boolean tracks whether the content has takes associated with it
+    val allContent: ObservableList<Pair<SimpleObjectProperty<Content>, SimpleBooleanProperty>>
+            = FXCollections.observableArrayList()
+    val filteredContent: ObservableList<Pair<SimpleObjectProperty<Content>, SimpleBooleanProperty>>
+            = FXCollections.observableArrayList()
+
+    // Whether the UI should show the plugin as active
+    private var showPluginActive: Boolean by property(false)
+    val showPluginActiveProperty = getProperty(ViewCollectionsViewModel::showPluginActive)
+
+
+    private var loading: Boolean by property(false)
+    val loadingProperty = getProperty(ViewCollectionsViewModel::loading)
+
+    // Create the use cases we need (the model layer)
+    private val accessTakes = AccessTakes(contentRepository, takeRepository)
+
+    init {
+        activeProjectProperty.toObservable().subscribe { setTitleAndChapters() }
+    }
+
+    private fun setTitleAndChapters() {
```

Comment:
"and" is a red flag in a method name. It indicates you're doing more than just one thing. 

here, you're binding the project title key (which, btw does the title key need to be looked up against messages?)
you're also binding the collections to your grid

these things are both distinct actions

Reply:
(see above as an example of how you could separate these out)

Reply:
cool. this is a function that matthew wrote. i just moved it over to a new file. I can make that change

Reply:
i think you might be right about the project title key. however, in the ui desings thomas has the book name in english and the language under it? i don't know if he wants the book name in english all the time?
@thomashajny

Reply:
Short answer: it would probably be nice if the book text wasn't always english. Its only useful if people can read it.

We will need to deal with how that affects the size of the text though. And we also need to determine what language the text gets translated to: Does it use the project language, the source language, or the system language?

Reply:
I'd imagine it would the target language? That's my initial feeling